geode-junit/src/main/java/org/apache/geode/test/util/ClassScanner.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.test.util;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ScanResult;
+
+/**
+ * This class is intended as a helper to the CI StressNewTest job.
+ */
+public class ClassScanner {
+
+  private ScanResult scanResult;
+  private String packageToScan;
+
+  public ClassScanner(String packageToScan) {
+    this.packageToScan = packageToScan;
+    scanResult = new ClassGraph().whitelistPackages(packageToScan)
+        .enableClassInfo()
+        .enableAnnotationInfo().scan();
+  }
+
+  public List<String> whatExtends(String classOrFile) {
+    Set<String> classesToConsider = new HashSet<>();
+    classesToConsider.add(pathToClass(classOrFile));
+
+    if (!classOrFile.contains(".")) {
+      classesToConsider.addAll(fullyQualifyClass(classOrFile));
+    }
+
+    return classesToConsider.stream()
+        .flatMap(x -> scanResult.getSubclasses(x).stream().map(ClassInfo::getSimpleName))
```

Comment:
Is this possibly missing running concrete classes that also have subclasses? Like someone extended a real test with more subclasses?

Reply:
In that case the selected subclass would end up executing tests in the super. A test could be missed if the subclass overrides a test in the super.